video/out/hwdec/hwdec_vaapi.c

```diff
@@ -359,6 +362,46 @@ static bool try_format_map(struct ra_hwdec *hw, struct mp_image *surface)
     return ok;
 }
 
+static bool try_format_upload(struct ra_hwdec *hw, int mp_fmt)
```

Comment:
This is not complex function, but I just noticed that we already have this functionality as `determine_working_formats()` in hwdec_vaapi. Maybe we should have it moved to some common places and use one implementation?

Reply:
What do you mean? This is already a part of `determine_working_formats`.

Reply:
ok, nevermind, I was looking at it backwards.